File: c/abc154_c_c.java
import java.util.Scanner;
import java.util.HashSet;
import java.util.Set;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Set<Double> hset = new HashSet<>();
        for (int i=0; i < n; i++) {
            Double ai = sc.nextDouble();
            hset.add(ai);
        }
        System.out.println(hset.size() == n ? "YES" : "NO");
    }
}

File: c/abc154_c_c.rs
use std::collections::HashSet;

fn main() {
    let n: usize = {
    let mut line = String::new();
    std::io::stdin().read_line(&mut line).unwrap();
    line.trim().parse().unwrap()
    };
    let a: Vec<i64> = {
        let mut line = String::new();
        std::io::stdin().read_line(&mut line).unwrap();
        line.split_whitespace()
            .map(|x| x.parse().unwrap())
            .collect()
    };
    let mut hset: HashSet<i64> = HashSet::new();
    for ai in a {
        hset.insert(ai);
    }
    let ans = if hset.len() == n {
        "YES"
    } else {
        "NO"
    };
    println!("{}", ans);
}